backend/app/repository/gitea_provider.py

```
@@ -0,0 +1,821 @@
+                Branch(
+                    name=branch["name"],
+                    protected=branch.get("protected", False),
+                    default=branch.get("default", False)
+                    or branch["name"] == default_branch_name,
+                ).model_dump()
+                for branch in all_branches
+            ]
+        except requests.exceptions.RequestException as e:
+            raise HTTPException(status_code=502, detail=f"Gitea API error: {str(e)}")
+
+    @lru_cache(maxsize=100)
```

Ruff (0.14.8)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)